lib/IMAP/MailboxSync.php

```diff
@@ -250,4 +268,36 @@ private function isMailboxShared(?Horde_Imap_Client_Namespace_List $namespaces,
 		}
 		return false;
 	}
+
+	private function syncMailboxStatus(mixed $mailboxes, ?string $personalNamespace, \Horde_Imap_Client_Socket $client): void {
+		[$sync, $doNotSync] = array_reduce($mailboxes, function (array $carry, Mailbox $mailbox) use ($personalNamespace) {
+			[$sync, $doNotSync] = $carry;
+			$inboxName = $personalNamespace === null ? "INBOX" : ($personalNamespace . $mailbox->getDelimiter() . "INBOX");
+			if ($inboxName === $mailbox->getName() || $mailbox->getSyncInBackground()) {
+				return [
+					[...$sync, $mailbox],
+					$doNotSync,
+				];
+			}
+			return [
+				$sync,
+				[...$doNotSync, $mailbox],
+			];
+		}, [[], []]);
+
+		// Synchronize the mailboxes selected for sync and keep the rest updated occasionally
+		shuffle($doNotSync);
+		/** @var Mailbox[] $syncStatus */
+		$syncStatus = [...$sync, ...array_slice($doNotSync, 0, 5)];
+		foreach ($syncStatus as $mailbox) {
+			$stats = $this->folderMapper->getFoldersStatusAsObject($client, $mailbox->getName());
```

Comment:
this is the only place we directly fetch the STATUS. @JohannesGGE please attach a debugger and see if this loop is responsible for the many STATUS commands, or if STATUS is requested somewhere else implicitly when we carry out another operation